modal/_resolver.py

```diff
@@ -101,14 +104,9 @@ async def load(self, obj: "_Object", existing_object_id: Optional[str] = None):
             # the same content
             cached_future = self._deduplication_cache.get(deduplication_key)
             if cached_future:
-
-                def hydrate_copy(fut):
-                    # in case an object is omitted due to content duplication, make sure the copy is
-                    # is still hydrated using hydration data from the original
-                    hydrated_object = fut.result()
-                    obj._hydrate(hydrated_object.object_id, self._client, hydrated_object._get_metadata())
-
-                cached_future.add_done_callback(hydrate_copy)
```

Comment:
This actually fixes a race bug, since the `hydrate_copy` callback wouldn't necessarily have run when an awaiter got back control after cached_future resolved, so if they then use the copy object it wouldn't be hydrated (!)